libraries/botbuilder-adapter-alexa/src/alexaAdapter.ts

```diff
@@ -0,0 +1,150 @@
+import {
+    BotAdapter,
+    ConversationReference,
+    TurnContext,
+    Activity,
+    ResourceResponse,
+    WebRequest,
+    WebResponse,
+    ActivityTypes
+} from 'botbuilder';
+import { RequestEnvelope, IntentRequest, Session, interfaces, SessionEndedRequest } from 'ask-sdk-model';
+import { getRequestType, getLocale, getUserId, createAskSdkError } from 'ask-sdk-core';
+import { retrieveBody } from '@botbuildercommunity/adapter-twitter';
+
+/**
+ * @module botbuildercommunity/adapter-alexa
+ */
+
+export class AdapterAlexa extends BotAdapter {
+    private responses: Map<string, Activity[]> = new Map<string, Activity[]>();
```

Comment:
Why save the responses in a `Map`? For what do we need them later on?

Reply:
the .net framework had this property. but something needs to be created to help the `processActivities` method and `sendActivities` methods to create an response for alexa. Alexa is expecting a their post to return a message back to the user. The botframework expects a simple 200 response with some uuid field and then sends the channel post in response.

Reply:
@iMicknl i have some ideas about how to break apart the adapters and the classes which convert the alexa api to botframework activity. let me know if you think something like this would be helpful: https://github.com/trashcanmonster8/botbuilder-community-js/tree/alexa-activity-class/libraries/botbuilder-adapter-alexa/src

I found it made testing a lot easier too.

Reply:
[My approach](https://github.com/BotBuilderCommunity/botbuilder-community-js/pull/95/files#diff-28cf8d8593fe3cf4ada24a4b00c276e4R61) is slightly different and based on the [BotKit BotBuilder adapters](https://github.com/howdyai/botkit/blob/master/packages/botbuilder-adapter-slack/src/slack_adapter.ts#L571). They use the `context.turnState.set('httpBody', body)` to send a response back. What do you think? 

I am not sure what the best way is for JavaScript adapters. I will have a look at the [WeChat Typescript adapter](https://github.com/microsoft/BotFramework-WeChat/pull/3), which is a work in progress. Let's see how they are tackling this.

Reply:
@iMicknl This is an interesting one. I think this might be where you see the differences between ported C# -> JS/TS (which I think is the case for the WeChat adapter) and those written in JS originally (i.e. BotKit, which we have just completed porting to C#).  My general steer here, at least for now, would be to follow the BotKit adapter patterns (they are certainly the ones that have had the most attention to date).  However, if you think the WeChat one is doing something better in places, then great.

Reply:
@garypretty agree. However for Twilio WhatsApp I had a look at the BotKit adapters and some features like attachment handling are not really aligned with the Bot Framework SDK. + the need for `restify.plugins.bodyParser()`, where this can easily be handled by some extra code.

I will have a look at the best practices from the BotKit adapters + the WeChat sample.

Reply:
@iMicknl Absolutely great with me.  Let's follow them if / where is makes sense, but if you think there is a better or more efficient way of doing things then great. 👍 

@iMicknl @trashcanmonster8 thanks to you both for your continued effort on this. It is awesome to see and a JS / TS version of the Alexa adapter is a common ask that we hear. I'll also apologies for my lack of familiarity with JS (it's on my list to ramp up on) 👍

Reply:
@trashcanmonster8 @iMicknl For the Twitter adapter, I followed @iMicknl's pattern of `context.turnState.set('httpBody', body)`. Botkit is a slightly different animal but Ben has some good patterns and practices we can utilize.